### PR TITLE
added permissionform file for issue Show readonly permissions in user edit…

### LINKED ISSUE #2556: Show readonly permissions in user edit page
Split from https://github.com/centerofci/mathesar/issues/2432

Refer Permissions block in [Figma UX](https://www.figma.com/file/xHb5oIqye3fnXtb2heRH34/Styling?node-id=4608%3A32284&t=YuvUELJVDoSAdXfB-1)

### PR DESCRIPTION
**Pull request by raunak-dev-edu**
… page

Fixes #2556

This Pull Request has partially solved the issue. So, making a draft PR.
**My workflow**
1] I have created a `UserPermissionForm.svelte` file in @mathesar/systems/users-and-permissions and used it as a component in the `EditUserPage.svelte`  file.
2] In UserPermissionForm, I have made a condition in which if the user having role will show the schema name with their role, and if not any permission, then a warning message will show up with `Please navigate to the appropriate database or schema in order to grant access to the user.`
3] For this, First, I import `Database` and `SchemaEntry` from @mathesar/AppTypes and make props with this, and for the user role I have import `UserModel` from @mathesar/stores/users, but when I try to access, then it shows me undefined. Like I have tried to access schema name as {schema.name} but an error arises as "cannot read properties of undefined (reading 'name')"
4] For warning messages, I use the `WarningBox` component that import it from @mathesar/components/message-boxes/WarningBox.svelte. This is working fine. 

**Screenshots**
Current work done 

image

Error-

image

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: mathesar_ui/src/systems/users-and-permissions/UserPermissionForm.svelte [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  import { Help } from '@mathesar-component-library';
+  import WarningBox from '@mathesar/components/message-boxes/WarningBox.svelte';
+  import type { UserModel } from '@mathesar/stores/users';
+  import { getSchemasStoreForDB } from '@mathesar/stores/schemas';
+  import { getUsersStoreFromContext } from '@mathesar/stores/users'; 
+  import type { Database, SchemaEntry } from '@mathesar/AppTypes';
+
+  export let database : Database;
+  export let schema : SchemaEntry;
+  export let userId: UserModel['id']
```
Comment (pavish) [inline mathesar_ui/src/systems/users-and-permissions/UserPermissionForm.svelte:11]: This component accepts database and schema but the values are not passed when calling the component instance in EditUserPage. So these values will be undefined.

Please check out [Svelte docs](https://svelte.dev/docs#template-syntax-attributes-and-props) on how to work with components and props.

File: mathesar_ui/src/systems/users-and-permissions/UserPermissionForm.svelte [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    <div class="user-without-permission">
+      <WarningBox fullWidth>
+        Please navigate to the appropriate database or schema in order to grant
+        access to the user.
+      </WarningBox>
+    </div>
+  {:else}
+  <div>
+    {database.name} / {schmema.name}
+  </div>
```
Comment (pavish) [inline mathesar_ui/src/systems/users-and-permissions/UserPermissionForm.svelte:33]: The values here should be based on the permissions of the user and not the current database and schema. They can be obtained by `databaseRoles` and `schemaRoles` properties in the UserModel and the required API calls need to be made to obtain the names of the schemas and dbs we don't have yet.

**Review by pavish — CHANGES_REQUESTED** [review]

File: mathesar_ui/src/systems/users-and-permissions/UserPermissionForm.svelte (hunk 2 above)
Comment (raunak-dev-edu) [reply mathesar_ui/src/systems/users-and-permissions/UserPermissionForm.svelte:33]: I am not able to get the schema name or database name from `schemaRoles` and `databaseRoles`. Could you please help me with how to make these api?

Comment (pavish) [reply mathesar_ui/src/systems/users-and-permissions/UserPermissionForm.svelte:33]: @raunak-dev-edu Figuring these details out is a major part of implementing this requirement. You're free to find another issue to work on if you find it hard to implement this.

**Pull request closed without merging**